src/span.cpp

```diff
@@ -154,6 +150,17 @@ void Span::FinishWithOptions(
     }
     // Don't erase the tag, in case it is populated with interesting information.
   }
+  tag = span_->meta.find(tags::analytics_event);
+  if (tag != span_->meta.end()) {
+    // tag->second is the JSON-serialized value of the variadic type given to SetTag.
+    // Apply only truthy or falsey values.
+    if (tag->second == "true" || tag->second == "1") {
```

Comment:
What if I do something like:
```cpp
span.SetTag(tags::analytics_event, 0.5);
```

While we don't want to explain this on our docs (very advanced usage), it's still something possible and part of our API. Will it increase the complexity of the code-block?

Reply:
This part is following the behavior of the [custom integrations](https://docs.datadoghq.com/tracing/trace_search_and_analytics/?tab=python#custom-instrumentation) which don't have rates, just true/false.
The C++ tracer doesn't have drop-in replacements for common packages like other languages have, where things can/should be controlled by rates.

If a user did to a `0.5`, for now it'd get ignored, which I think is consistent with other implementations.

Reply:
Oh, all integrations (and tracers) have rates. They are just not documented (because we don't suggest to do that): https://github.com/DataDog/dd-trace-py/blob/master/tests/test_span.py#L252-L260. It's explained also in the RFC (we should start making them public!).

So if you set `0.5` instead of `true`, that value must be set as metric, and nothing else needs to be done. So to be consistent with other tracers, we must support this case.